packages/core/test/util/derivetype.test.ts

```diff
@@ -0,0 +1,101 @@
+/*
+  The MIT License
+  
+  Copyright (c) 2019 EclipseSource Munich
+  https://github.com/eclipsesource/jsonforms
+  
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+  
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import test from 'ava';
+
+import { JsonSchema } from '../../src/models/jsonSchema';
+import { deriveType } from '../../src/util/index';
+
+test('derive type with type', t => {
+  const schema: JsonSchema = {
+    type: 'string'
+  };
+  t.is(deriveType(schema), 'string');
+});
+
+test('derive type w/o type - properties object', t => {
+  const schema: JsonSchema = {
+    properties: {
+      foo: { type: 'string' }
+    }
+  };
+  t.is(deriveType(schema), 'object');
+});
+
+test('derive type w/o type - additionalProperties object', t => {
+  const schema: JsonSchema = {
+    additionalProperties: {
+      type: 'string'
+    }
+  };
+  t.is(deriveType(schema), 'object');
+});
+
+test('derive type w/o type - items array', t => {
+  const schema: JsonSchema = {
+    items: {
+      type: 'string'
+    }
+  };
+  t.is(deriveType(schema), 'array');
+});
+
+test('derive type with type - union', t => {
+  const schema: JsonSchema = {
+    type: ['string', 'number']
+  };
+  t.is(deriveType(schema), 'string');
+});
+
+test('derive type with type - union', t => {
```

Comment:
What the difference to the previous test?

Reply:
that is a wrong copy, thank you for finding this